### ISSUE TITLE
StatsModels.DataFrameRegressionModel should be StatsModels.TableRegressionModel

### ISSUE THREAD

**Issue by arnavs** [jmboehm/RegressionTables.jl#26]
This doesn't exist anymore. It's now called `TableRegressionModel.` See, e.g., what happens when you run the first example [here](https://github.com/JuliaStats/GLM.jl/blob/b2477efef78d3748b01189724838aa5da0a2faed/docs/src/examples.md): 

```
julia> data = DataFrame(X=[1,2,3], Y=[2,4,7])
3×2 DataFrame
│ Row │ X     │ Y     │
│     │ Int64 │ Int64 │
├─────┼───────┼───────┤
│ 1   │ 1     │ 2     │
│ 2   │ 2     │ 4     │
│ 3   │ 3     │ 7     │

julia> ols = lm(@formula(Y ~ X), data)
StatsModels.TableRegressionModel{LinearModel{GLM.LmResp{Array{Float64,1}},GLM.DensePredChol{Float64,LinearAlgebra.Cholesky{Float64,Array{Float64,2}}}},Array{Float64,2}}
```

**Note**: This issue currently causes precompilation to fail.

**Comment by jlperla** [comment]
Hi Johannes!  We would love to push your  package for teaching this fall, but we are having some incompatibility problems.  @matthieugomez  take note of this for now as well as we ran into this with FixedEffects.

**Comment by jmboehm** [comment]
Hi Jesse,
I hope to get to it in the next few days. It's the first week of term and things are a bit crazy...

**Comment by jlperla** [comment]
Of course.  If we are able to see if we can work on a PR in the meantime, we will give you an update

**Comment by jlperla** [comment]
@jmboehm  Would you like us to take a crack at the PR for this?

**Comment by jmboehm** [comment]
Thanks for the offer. I can have a go at it this weekend. If you want/need it earlier, please go ahead.

**Comment by jlperla** [comment]
@jmboehm  : we gave it a shot but failed to converge on a solution quickly.  We suspect that familiarity with the codebase might make it a lot easier.  If you are unable to do it, we can come back to it.  No pressure or rush from us.

**Comment by jmboehm** [comment]
Should be fixed by #28 (as long as you are on julia >= 1.1). Let me know if you run into issues.

**Comment by jlperla** [comment]
Thanks so much.  After the release is tagged we will update our manifests and reenable it in the quantecon lecture notes.
